### DISCUSSION TITLE
Alt text for media

### DISCUSSION THREAD

**Discussion by lo2dev** [stoatchat/discussions#664] [Feature Suggestions]
A similar [issue](https://github.com/orgs/revoltchat/discussions/584) has been made a while ago but it seems to be focused on codeblocks and not media so I'm writing this separately.

Currently you can't set alt text for images or videos, but it would greatly benefit people with disabilities that use Revolt. My reference for alt text implementation is the fediverse as I don't use any other chat app.

Frontend solutions:

In the pre-upload phase, the attachments should have the following buttons: spoiler toggle (but that's another issue), close button and the alt text button which spawn a modal so the user can input alt text. Here's a rough concept I made in GTK to illustrate (of course no one has to follow these one to one):
![screenshot showcasing attachments actions before uploading them to revolt](https://github.com/revoltchat/revolt/assets/115636509/f6d2ce60-7b4e-48d1-9313-b9094a5efa7d)
My only gripe with this is that there might be some issues on mobile interfaces.

The alt modal should feature the image and a text field. The image should be big enough to be seen fully  and the text field big enough so it doesn't feel cramped in case of big alt text. Concept in GTK:
![image showcasing a GTK window for adding alt text to media](https://github.com/revoltchat/revolt/assets/115636509/4c53f99a-72ee-42df-9b13-3d0edc366b5d)

Images on chat should have a little button in their corner with the label "ALT" which spawns a modal with the text selectable. 

Backend solutions:

I am not knowledgeable enough in this regard, but I would guess there would be two new API changes. One to set the alt text related to the specific image and one to get the alt text.

**Comment by aisuleep** [comment]
If/When this gets implemented, it would be good if it had an indicator noting the amount of characters left the description can have.
